pumpkin-inventory/src/window_property.rs

```diff
@@ -0,0 +1,103 @@
+pub trait WindowPropertyTrait: Sized {
+    fn to_id(self) -> i16 {
+        0
```

Comment:
I think this should be `self as i16`. Then remove all the other implementations that are this.

Reply:
I tried doing that, but we'd have to constrain the trait further than just Sized in that case. Do you know what supertrait is for `enum as u8`?

Reply:
I'm not around a computer right now sorry, but I'm pretty sure you just put `#[repr(u8)]` on the Enum, then hit it with the Enum::from_u8

Reply:
Why do you need sized at all? In fact, why have an explicit type at all for windowproperties? Just implement to_packet on the WindowPropertyTrait.

To get the conversion you just need to restrict the bounds to an Into<u8> or something similar.

Reply:
It is Sized so that we can take self instead of &self in the trait